source/common/http/http1/codec_impl.cc

```diff
@@ -57,6 +57,42 @@ HeaderKeyFormatterPtr formatter(const Http::Http1Settings& settings) {
 
   return nullptr;
 }
+
+/**
+ * Determines whether Envoy supports the encodings provided for the transfer-encoding header. Envoy
+ * currently only supports chunked and identity. Per RFC 7230 Section 3.3.3 when encodings are
+ * chained then chunked must be the last encoding, so possible values are: "chunked",
+ * "identity,chunked" "identity, chunked", etc.
+ *
+ * @param encoding The transfer-encoding value from the HTTP message
+ * @return whether the encoding only contains supported encodings.
+ */
+bool IsSupportedTransferEncoding(absl::string_view encoding) {
```

Comment:
I'd prefer we not change this behavior in this PR.
Updating htttp parser allows to make this change, but it's really helpful to keep behavioral changes isolated where possible so folks can bisect if there's a problem etc.

Can we do the pure upgrade, and do this as a follow-up?

Reply:
+1

Reply:
ack, doing just the upgrade for now. The upgrade itself contains a change that removes support for just `transfer-encoding: identity` which was previously supported. Do you want me to figure out how to retain that behavior? That _may_ involve running the parser in "lenient" mode and adding more codec code to cover the checks the parser used to cover.